bob/bob.go

```diff
@@ -1,15 +1,37 @@
-// This is a "stub" file.  It's a little start on your solution.
-// It's not a complete solution though; you have to write some code.
-
-// Package bob should have a package comment that summarizes what it's about.
-// https://golang.org/doc/effective_go.html#commentary
 package bob
 
-// Hey should have a comment documenting it.
+import (
+	"regexp"
+	"strings"
+)
+
+// Hey function which responds correctly
 func Hey(remark string) string {
-	// Write some code here to pass the test suite.
-	// Then remove all the stock comments.
-	// They're here to help you get started but they only clutter a finished solution.
-	// If you leave them in, reviewers may protest!
-	return ""
-}
+	remark = strings.Trim(remark, " \t\n\r")
+
+	re, _ := regexp.Compile("[a-zA-Z]+")
```

Comment:
With regex it's generally good practice not to compile them at runtime. The reasons for this being that they're not particularly quick to compile and also it's not something that is likely to change.

Have a look at the `regexp.MustCompile` function, this is something that you can call outside of function iteself, using it to declare a package level variable. When you've done this compare the benchmark resuult with what you had before.

Reply:
As an extension to this you might also want to look at the `MatchString` method in the regexp package, this is a method available on a compiled regexp object. Again checkout the benchmark results vs using `FindAllString`